### ISSUE TITLE
libtiff closes fp prematurely after n_frames, causing seek to fail

### ISSUE THREAD

**Issue by lambdafu** [python-pillow/Pillow#2229]
### What did you do?

```python
    TiffImagePlugin.READ_LIBTIFF = True
    with Image.open("/tmp/multipage.tiff") as im:
        for idx in range(im.n_frames):
                im.seek(idx)
                im.load()
```

### What did you expect to happen?

The frames should be loaded one after another.

### What actually happened?

Loading the second (and any further) frame fails with ValueError('I/O operation on closed file',).

### What versions of Pillow and Python are you using?

17d38d09050ea6a546fcb3ddfce189c2b82784fe

**Comment by wiredfool** [comment]
Is this a regression due to #2225?

**Comment by lambdafu** [comment]
Nope, this is an old bug., probably as old as the introduction of n_frames.

**Comment by wiredfool** [comment]
Fixed in #2330
